pgcontents/hybridmanager.py

```diff
@@ -281,3 +246,44 @@ def delete(self, path):
         'checkpoint_id',
         False,
     )
+
+    # CODE WRITTEN BY VIADUCT
+    @outside_root_to_404
+    def rename(self, old_path, new_path):
+        """Ensure that roots of our managers can't be moved or renamed.
+
+        This a precaution against rogue behavior
+        """
+        old_prefix, old_mgr, old_mgr_path = _resolve_path(
+            old_path, self.managers)
+
+        new_prefix, new_mgr, new_mgr_path = _resolve_path(
+            new_path,
+            self.managers,
+        )
+        # do not allow moving/renaming the root
+        if old_mgr_path in self.managers or new_mgr_path in self.managers:
+            raise HTTPError(
+                400, "Can't rename or move root directories ({old} -> {new})".
+                format(
+                    old=old_path,
+                    new=new_path,
+                ))
+
+        if old_mgr is not new_mgr:
+
+            # get the model from the old_mgr
+            model = old_mgr.get(old_mgr_path)
+
+            # save the model with the new_mgr
+            new_mgr.save(model, new_mgr_path)
+
+            # delete the model with the old_mgr using the protected delete
+            self.delete(old_mgr_path)
+
+            # Return
+            return None
```

Comment:
is `None` the correct return type? (will `new_mgr.rename(old_mgr_path, new_mgr_path)` return `None` as well?)

Reply:
Looking at other contentmanagers, this appears to be OK